### PR TITLE
Makes tests 🍏

### PR DESCRIPTION
**Pull request by ashfurrow**
Still to do before merging into master:

- Code golfing to remove references to RAC :golf: 
- Thorough testing :warning:

### CODE REVIEW

File: Kiosk/Bid Fulfillment/RegistrationPasswordViewModel.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ class RegistrationPasswordViewModel: RegistrationPasswordViewModelType {
             .sharedProvider
             .request(ArtsyAPI.FindExistingEmailRegistration(email: email))
             .map(responseIsOK)
-            .replay(1)
+            .shareReplay(1)
```
Comment (orta) [inline Kiosk/Bid Fulfillment/RegistrationPasswordViewModel.swift]: what's different here?

File: Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift [hunk 2]
```diff
@@ -38,6 +39,25 @@ class SaleArtworkDetailsViewController: UIViewController {
...
+        dispatch_async(dispatch_get_main_queue()) {
+            self.layoutSubviews.onNext()
+        }
+    }
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+
+        viewWillAppear.onCompleted()
+    }
```
Comment (orta) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: doubt this is still needed

Comment (orta) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: I'm gonna hold off on a merge for this one.

Comment (ashfurrow) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: Yeah, I actually went through and this line indicates intent more clearly. The other jiggery-pokery (`self.view.setNeedsLayout` etc) actually _does_ need to be there to work :shrug:

File: Kiosk/Bid Fulfillment/RegistrationPasswordViewModel.swift (hunk 1 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/RegistrationPasswordViewModel.swift]: Yup!

**Comment by ashfurrow** [comment]
Tests are passing locally, from Xcode and from the command line. We'll see...

**Comment by ashfurrow** [comment]
screen shot 2015-12-01 at 6 11 11 pm

> value of type 'Self' has no member 'ignoreElements'

This is dumb, Travis. Stop being dumb.

**Comment by alloy** [comment]
How is that Travis’ fault?

**Comment by ashfurrow** [comment]
It's giving a compiler error that methods are missing, but those methods do exist in RxSwift (beta 3, not beta 2). I think it might be caching the library somehow, need to investigate.

**Comment by alloy** [comment]
Aha, thanks for clarifying.

**Comment by ashfurrow** [comment]
:green_heart:

File: Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift (hunk 2 above)
Comment (orta) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: Alright!

File: Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift [hunk 3]
```diff
@@ -371,8 +385,10 @@ class SaleArtworkDetailsViewController: UIViewController {
         retrieveAdditionalInfo()
             .filter { info in
                 return info.isNotEmpty
-            }.subscribeNext { info in
+            }.subscribeNext { [weak self] info in
                 additionalInfoLabel.attributedText = MarkdownParser().attributedStringFromMarkdownString(info)
+                self?.view.setNeedsLayout()
+                self?.view.layoutIfNeeded()
```
Comment (orta) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: then are you sure this is needed?

Comment (ashfurrow) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: Yup.

Comment (orta) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: this is the only blocker. If this is cool, then merge away.

Comment (ashfurrow) [inline Kiosk/Sale Artwork Details/SaleArtworkDetailsViewController.swift]: :tada:

**Pull request merged**